packages/lit-html/src/directives/keyed.ts

```diff
@@ -0,0 +1,37 @@
+import {nothing} from '../lit-html.js';
+import {
+  directive,
+  Directive,
+  ChildPart,
+  DirectiveParameters,
+} from '../directive.js';
+import {setCommittedValue} from '../directive-helpers.js';
+
+class Keyed extends Directive {
+  key: unknown = nothing;
+
+  render(k: unknown, v: unknown) {
+    this.key = k;
+    return v;
+  }
+
+  override update(part: ChildPart, [k, v]: DirectiveParameters<this>) {
+    if (k !== this.key) {
+      // Clear the part before returning a value. This forces a re-render.
+      setCommittedValue(part);
```

Comment:
Man, had to look up the API for what no 2nd argument does. I remember now we defaulted it to a RESET_VALUE sigil that ensures any dirty check will fail. Maybe clarify in the comment that omitting the 2nd arg is what forces a re-render.

Reply:
done